### PR TITLE
Feat/gho safety module stake [skip cypress]

### PR DESCRIPTION
**Pull request by foodaka**
## General Changes

- Adds support for stkGHO staking

## Developer Notes

Add any notes here that may be helpful for reviewers.
- Needs to be on specific fork 
```
NEXT_PUBLIC_FORK_BASE_CHAIN_ID=1
NEXT_PUBLIC_FORK_CHAIN_ID=1
NEXT_PUBLIC_FORK_URL_RPC="https://rpc.tenderly.co/fork/77ef1d83-2d75-4822-bd87-a650dc54f5b9"
```

## Author Checklist

Please ensure you, the author, have gone through this checklist to ensure there is an efficient workflow for the reviewers.

- [ ]  The base branch is set to `main`
- [ ]  The title is using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) formatting
- [ ]  The Github issue has been linked to the PR in the Development section
- [ ]  The General Changes section has been filled out
- [ ]  Developer Notes have been added (optional)

**If the PR is ready for review:**

- [ ]  The PR is in `Open` state and not in `Draft` mode
- [ ]  The `Ready for Dev Review` label has been added

## Reviewer Checklist

Please ensure you, as the reviewer(s), have gone through this checklist to ensure that the code changes are ready to ship safely and to help mitigate any downstream issues that may occur.

- [ ]  End-to-end tests are passing without any errors
- [ ]  Code style generally follows existing patterns
- [ ]  Code changes do not significantly increase the application bundle size
- [ ]  If there are new 3rd-party packages, they do not introduce potential security threats
- [ ]  If there are new environment variables being added, they have been added to the `.env.example` file as well as the pertinant `.github/actions/*` files
- [ ]  There are no CI changes, or they have been approved by the DevOps and Engineering team(s)
- [ ]  Code changes have been quality checked in the ephemeral URL
- [ ]  QA verification has been completed
- [ ]  There are two or more approvals from the core team
- [ ]  Squash and merge has been checked

### CODE REVIEW

**Comment by MareskoY** [comment]
**Bug 1** cooldown unstake for GHO doesn't work - unstake not activated
cooldown


**Bug 2** GET GHO button doesn't work
Screenshot 2024-01-17 at 23 59 51

Also 
1. Staking APR is 0 not possible to check reward
2. pr outdated from main

File: src/ui-config/stakeConfig.ts [hunk 1]
```diff
@@ -8,22 +9,30 @@ export interface StakeConfig {
...
       STAKING_HELPER?: string;
+      TOKEN_ORACLE: string;
     };
   };
 }
 
 export const stakeConfig: StakeConfig = {
   chainId: ChainId.mainnet,
-  stakeDataProvider: '0x5E045cfb738F01bC73CEAFF783F4C16e8B14090b',
+  stakeDataProvider: '0x972377456C7ae9C070416d0234303387A14e6972', // NOTE FORKED v1
```
Comment (foodaka) [inline src/ui-config/stakeConfig.ts:19]: PROD Address Book

File: src/ui-config/stakeConfig.ts [hunk 2]
```diff
@@ -8,22 +9,30 @@ export interface StakeConfig {
...
     },
     [Stake.bpt]: {
-      TOKEN_STAKING: '0xa1116930326D21fB917d5A27F1E9943A9595fb47',
-      STAKING_REWARD_TOKEN: '0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9',
+      TOKEN_STAKING: AaveSafetyModule.STK_ABPT,
+      STAKING_REWARD_TOKEN: AaveV3Ethereum.ASSETS.AAVE.UNDERLYING,
+      TOKEN_ORACLE: AaveSafetyModule.STK_ABPT_ORACLE,
+    },
+    [Stake.gho]: {
+      TOKEN_STAKING: '0x71c244817537d68a990038669efb4f68aca76a8f', // '0xcC12d8ed9111cA6a99BE5ded6F9CD6f21C360502', // TODO stkGHO FORK
```
Comment (foodaka) [inline src/ui-config/stakeConfig.ts:33]: PROD ADDRESS BOOK

File: src/ui-config/stakeConfig.ts [hunk 3]
```diff
@@ -8,22 +9,30 @@ export interface StakeConfig {
...
+      STAKING_REWARD_TOKEN: AaveV3Ethereum.ASSETS.AAVE.UNDERLYING,
+      TOKEN_ORACLE: '0x3f12643d3f6f874d39c2a4c9f2cd6f2dbac877fc', // TODO GHO ORACLE ADDRESS BOOK
```
Comment (foodaka) [inline src/ui-config/stakeConfig.ts:35]: PROD ADDRESS BOOK

File: src/ui-config/stakeConfig.ts [hunk 4]
```diff
@@ -8,22 +9,30 @@ export interface StakeConfig {
...
   chainId: ChainId.mainnet,
-  stakeDataProvider: '0x5E045cfb738F01bC73CEAFF783F4C16e8B14090b',
+  stakeDataProvider: '0x972377456C7ae9C070416d0234303387A14e6972', // NOTE FORKED v1
   tokens: {
     [Stake.aave]: {
-      TOKEN_STAKING: '0x4da27a545c0c5b758a6ba100e3a049001de870f5',
-      STAKING_REWARD_TOKEN: '0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9',
+      TOKEN_STAKING: AaveSafetyModule.STK_AAVE,
+      STAKING_REWARD_TOKEN: AaveV3Ethereum.ASSETS.AAVE.UNDERLYING,
       STAKING_HELPER: '0xce0424653fb2fd48ed1b621bdbd60db16b2e388a',
```
Comment (foodaka) [inline src/ui-config/stakeConfig.ts:24]: PROD ADDRESS BOOK

File: pages/staking.staking.tsx [hunk 5]
```diff
@@ -76,23 +103,22 @@ export default function Staking() {
...
+      ), // "0"
+    18 + 8
   );
 
   // Total AAVE Emissions (stkaave dps + stkbpt dps)
   const stkEmission = formatEther(
-    BigNumber.from(stakeGeneralResult?.aave.distributionPerSecond || '0')
-      .add(stakeGeneralResult?.bpt.distributionPerSecond || '0')
+    BigNumber.from(stkAave?.distributionPerSecond || '0')
+      .add(stkBpt?.distributionPerSecond || '0')
```
Comment (grothem) [inline pages/staking.staking.tsx:121]: we should add the stkGho emissions in here as well

File: pages/staking.staking.tsx [hunk 6]
```diff
@@ -160,6 +191,28 @@ export default function Staking() {
...
+                  maxSlash="1" // 100%
+                  icon="gho"
+                  stakeData={stkGho}
+                  stakeUserData={stkGhoUserData}
+                  ethPriceUsd={stakeGeneralResult?.ethPriceUsd}
+                  onStakeAction={() => openStake('gho', 'gho')}
+                  onCooldownAction={() => openStakeCooldown('gho')}
+                  onUnstakeAction={() => openUnstake('gho', 'gho')}
+                  onStakeRewardClaimAction={() => openStakeRewardsClaim('gho', 'AAVE')}
+                  headerAction={<GetGhoToken />}
```
Comment (grothem) [inline pages/staking.staking.tsx:212]: should we just remove this header for now?

File: pages/staking.staking.tsx [hunk 7]
```diff
@@ -160,6 +191,28 @@ export default function Staking() {
...
+              >
+                <StakingPanel
+                  stakeTitle="GHO"
+                  stakedToken="GHO"
+                  maxSlash="1" // 100%
+                  icon="gho"
+                  stakeData={stkGho}
+                  stakeUserData={stkGhoUserData}
+                  ethPriceUsd={stakeGeneralResult?.ethPriceUsd}
+                  onStakeAction={() => openStake('gho', 'gho')}
```
Comment (grothem) [inline pages/staking.staking.tsx:208]: Should be 
```
openStake('gho', 'GHO')
```
The second param is what shows in the action buttons for the text, so all caps will be consistent with the others. Same thing with the `openUnstake` method

File: pages/staking.staking.tsx (hunk 5 above)
Comment (foodaka) [reply pages/staking.staking.tsx:121]: added c7a1a9f3553716232fca3f99c11e5debcb88d82a

**Review by grothem — APPROVED** [review]

**Review by MareskoY — APPROVED** [review]

**Review by miguelmtzinf — APPROVED** [review]

**Pull request merged**